### PR TITLE
Add wildcard analyzer

### PR DESCRIPTION
**Pull request by MBkkt**
- [x] Analyzer description
- [x] View examples
- [x] Inverted index examples
- [x] Explain how the search works internally?
- [x] Release notes

#### Upstream PRs

- 3.10:
- 3.11:
- 3.12: arangodb/enterprise-preview:devel-nightly

### CODE REVIEW

**Comment by Simran-B** [comment]
The Analyzer seems to work differently than the one in Elastic, at least based on what they show in their blog post. They don't seem to create all of the n-grams, e.g. for `avocado` and n-gram size = 3, they would apparently add `avo`, `oca`, `ado` to the index but our Analyzer returns:

`av`, `avo`, `voc`, `oca`, `cad`, `ado`, `do`, `o`.

The first token seems to always be a prefix with `ngramSize - 1`? Then all the trigrams, and I suppose the rest of the strings down to a size of 1 are for suffix matching. I don't quite understand the first entry. If ngramSize is e.g. 5 but the input is shorter than this, the first two entries are identical. Is this intentional?

**Comment by MBkkt** [comment]
@Simran-B 

It returns all ngram with specified `ngramSize`, and all suffix ngrams smaller than `ngramSize`

It also change input text to the `\xFFtext\xFF` -- \xFF invalid utf-8 and cannot appear in any text, so it real output is
`\xFFav, avo, voc, oca, cad, ado, do\xFF, o\xFF`

This is necessary to speedup queries with max sub pattern size smaller than ngramSize and prefix/suffix queries

**Comment by MBkkt** [comment]
@Simran-B 

> The Analyzer seems to work differently than the one in Elastic, at least based on what they show in their blog post. They don't seem to create all of the n-grams, e.g. for avocado and n-gram size = 3, they would apparently add avo, oca, ado to the index but our Analyzer returns:

It's just simplification of blog-post
If analyzer will produce for `avocado` only `avo oca ado`.
It will imposible to fast find it with query like `%cad%`

But it's possible on search phase avoid all tokens which intersects.
So if you honestly ngram `avocado`, it's possible to search it via `avo cad o`
It can be better in some cases, but from our measurements commonly it slower (because smaller count of distinct terms in approximation query)

File: site/content/3.12/index-and-search/analyzers.md [hunk 1]
```diff
@@ -1048,6 +1050,83 @@ db._query(`LET str = 'Test\twith An_EMAIL-address+123@example.org\n蝴蝶。\u20
...
+  a slower search (for `ref%` with post-validation using the ICU regular expression
+  engine)
+- `analyzer` (object, _optional_): an Analyzer definition-like objects with
+  `type` and `properties` attributes
+
+You cannot set the `offset` [Analyzer features](#analyzer-features) for this
+Analyzer. You can create the Analyzer without features, or use
+`["frequency", "position"]` which uses more memory but doesn't need to read
+stored column values for prefix, suffix, infix, and exact queries (`something%`,
+`%something`, `%something%`, `something`, but not `some%thing` or `%some%thing%`)
```
Comment (MBkkt) [inline site/content/3.12/index-and-search/analyzers.md:1085]:
```suggestion
`%something`, `%something%`, `something`, but not `some%thing` or `_something`)
```

**Review by MBkkt — COMMENTED** [review]
lgtm

**Comment by Simran-B** [comment]
The API change described in #447 broke an example. The example erroneously tried to drop a collection that is part of a graph but dropping the example graph drops all graph collections anyway.

Another issue that surfaced is that curl examples that purposefully trigger an error need to make use of `// xpError(...)` or the new toolchain complains about an unexpected error. In the old toolchain, the correct example behavior was only ensured by `assert()` statements.

**Review by nerpaula — APPROVED** [review]
LGTM

**Pull request merged**